plugins/modules/proxmox_cluster_ha_resources.py

```diff
@@ -148,13 +148,16 @@ def create(self, resources, sid, comment, group, max_relocate, max_restart, stat
             "state": state,
         }
 
+        # Normalize input SIDs to version without : in them
+        if ':' in sid:
+            sid = sid.split(':')[1]
+
         for resource in resources:
-            if resource["sid"] != sid:
+            if resource["sid"].split(':')[1] != sid:
                 continue
 
-            if ((resource.get("comment", ""), resource.get("group", ""), resource.get("max_relocate", 1),
-                 resource.get("max_restart", 1), resource["state"]) ==
-                    (comment, group, max_relocate, max_restart, state)):
+            # Compare all keys in the desired state against current state
+            if all(resource.get(k) == v for k, v in data.items()):
```

Comment:
this may not work for optional properties. 
e.g. comment may be missing in resources, it's "" in data - could be considered "equal"

Reply:
My case is different:
The service response for the object lacks the comment attribute as it is not set.
The desired state does not specify comment, it defaults to "".

```
resource.get("comment") == None <-> state["comment"] == ""
```

It does not need an update?

Reply:
Not from what I was able to observe :-)